### ISSUE TITLE
Make .dio.svg content diff friendly

### ISSUE THREAD

**Issue by Speedy37** [hediet/vscode-drawio#119]
Currently the draw.io xml data is stored in base64 format within the `content` attribute of the `svg` element.

It would be nice to have an option to store this data directly in xml (svg is xml) within the svg.

**Comment by hediet** [comment]
Nice idea! However, I think this should
1) either be implemented in draw.io itself (maybe we can open an issue in their repo?)
2) or with forward compatibility so that the web app can still open such a file. This could be achieved by keeping the content attribute as a fall back, along your proposed metadata elemetn.

I want to avoid creating an incompatible format. Also, are you sure that no SVG processing tool interprets your proposed `metadata` element?

**Comment by Speedy37** [comment]
Upstreaming changes to the encoding/decoding step to draw.io seems the best way to proceed. 

Changes should be minimal and would only requires a new option to the message API.

Depending on how my free time goes this week, I might find some time to allocate experimenting with this approach.

**Comment by hediet** [comment]
Did you find time / manage to get something working? ;)

**Comment by Speedy37** [comment]
Not yet

**Comment by hediet** [comment]
Btw. if you have paypal or github sponsors, I can share some of the amount I get from github sponsors with you for your work on this extension ;)

**Comment by Speedy37** [comment]
Thx for the proposition, but for now this is a hobby :)

**Comment by septatrix** [comment]
I agree in that this is an upstream issue. However this is probably more involved than you think as SVG does not support any element (to my knowledge). Therefore a new namespace would have to be defined and declared. What I am doing currently is saving the File as uncompressed XML which seems to do the job just fine.
